src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -581,55 +453,8 @@ private void VerifyCanCreateOperationRequestMessage(bool synchronousCall, string
         /// <param name="uri">The Uri to be used for this request operation.</param>
         /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
         /// <returns>The message that can be used to write the request operation.</returns>
-        private ODataBatchOperationRequestMessage CreateOperationRequestMessageImplementation(string method, Uri uri, string contentId)
-        {
-            if (this.changeSetBoundary == null)
-            {
-                this.InterceptException(this.IncreaseBatchSize);
-            }
-            else
-            {
-                this.InterceptException(this.IncreaseChangeSetSize);
-            }
-
-            // write pending message data (headers, response line) for a previously unclosed message/request
-            this.WritePendingMessageData(true);
-
-            // Add a potential Content-ID header to the URL resolver so that it will be available
-            // to subsequent operations.
-            // Note that what we add here is the Content-ID header of the previous operation (if any).
-            // This also means that the Content-ID of the last operation in a changeset will never get
-            // added to the cache which is fine since we cannot reference it anywhere.
-            if (this.currentOperationContentId != null)
-            {
-                this.urlResolver.AddContentId(this.currentOperationContentId);
-            }

```

Comment:
This method should be private and call into abstract method implemented in derived class.